pkg/asset/cluster/tfvars.go

```diff
@@ -940,7 +940,24 @@ func (t *TerraformVariables) Generate(parents asset.Parents) error {
 			}
 		}
 
-		transitGatewayEnabled := powervsconfig.TransitGatewayEnabledZone(installConfig.Config.Platform.PowerVS.Zone)
+		attachedTG := ""
+		tgConnectionVPCID := ""
+		if installConfig.Config.PowerVS.ServiceInstanceGUID != "" {
+			attachedTG, err = client.GetAttachedTransitGateway(ctx, installConfig.Config.PowerVS.ServiceInstanceGUID)
+			if err != nil {
+				return err
+			}
+			if attachedTG != "" && vpcName != "" {
+				vpc, err = client.GetVPCByName(ctx, vpcName)
```

Comment:
Any reason not to use the one you got from the previous call?

Reply:
agreed, should be good to use the one from the previous call, that's why the scope of `vpc` was moved above